lightbulb/command_handler.py

```diff
@@ -794,7 +794,7 @@ class useful for extracting the arguments from the raw string and the property
             conv = converters.pop(0)
             arg_name = arg_names.pop(0)
 
-            if not isinstance(conv, _DefaultingConverter) and not arg_string:
+            if not isinstance(conv, (_DefaultingConverter, _GreedyConverter)) and not arg_string:
```

Comment:
If there is no default value given then Greedy requires **at least** one value so this behaviour is incorrect

Reply:
I see, I think we can allow it only if `unpack` was set to True, which means it's a var pos arg

Reply:
commit 1ca19314ca3d8fcb7bf96302c21be08174804078 makes it so that:

Case 1: `async def x(ctx: Context, args: Greedy[int]): ...`
- `x 1` and `x 1 a` results in `[1]`
- `x a` results in `[]` which then raises ConverterFailure
- `x` raises NotEnoughArguments

Case 2: `async def x(ctx: Context, *args: int): ...`
- `x 1` and `x 1 a` results in `(1,)`
- `x a` and `x` results in `()`

Case 3: `async def x(ctx: Context, args: Greedy[int] = [0]: ...`
- `x 1` and `x 1 a` results in `[1]`
- `x a` and `x` results in `[0]`

Is this the behaviour you were intending to produce? I personally would like the old behaviour since if it allows `[]`, the user has 100% control over it, they can re-raise the error themselves if they wanted it to have at least 1 element in it. But if doesn't as it raises the error in the first place, it's kinda hacky to make it allow `[]`, setting a default value isn't a solution. Having a mutable default argument is evil.

Reply:
Ok yeah the previous behaviour should be fine then, not sure that python even allows you to have mutable defaults anyway